### PR TITLE
Search this area 

### LINKED ISSUE #591: "Search this area" to launch a nearby search around that location
I often plan my itinerary and decide where to go, before actually being there.

On the Nearby map, it would be great if I could long-press a location to launch a nearby search around there.

### PR DESCRIPTION
**Pull request by neslihanturan**
**Description (required)**

Fixes #591 "Search this area" to launch a nearby search around that location

What changes did you make and why?
- When you moved on the map a button "Search this area" appears on top of the map
- If you don't click on the button, and come back your previous position, button disappears
- If you click to the button, a progress bar will appear on the map, and map will be locked, you can not move on the map during fetching places
- Only your location marker and your new search results will be visible on map and list
- If you move, map won't follow you if search this area mode on
- If you click to extend list, map camera won't change its target
- When you click on recenter me button, or go back to previous position on map with your fingers, nearby places around you will be uploaded
- Map will follow you as always
- Extending list will change map camera angle, so that your location will still be visible

**Tests performed (required)**
Tested betaDebug, on API19 and 26
**Screenshots showing what changed (optional - for UI changes)**

image

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/nearby/NearbyFragment.java [hunk 1]
```diff
@@ -312,6 +319,33 @@ private void refreshView(LocationServiceManager.LocationChangeType locationChang
         }
     }
 
+    public void refreshViewForCustomLocation(LatLng customLatLng) {
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/nearby/NearbyFragment.java]: Could we get Javadocs for these new methods, please?

File: app/src/main/java/fr/free/nrw/commons/nearby/NearbyMapFragment.java [hunk 2]
```diff
@@ -323,7 +347,50 @@ private void updateMapToTrackPosition() {
...
+        }
+    }
+
+    /**
+     * Updates camera position according to list sheet status. If list sheet is collapsed, camera
+     * focus should be in the center. If list sheet is expanded, camera focus should be visible
+     * on the gap between list sheet and tab layout.
+     * @param isBottomListSheetExpanded
+     */
+    private void updateMapCameraAccordingToBottomSheet(boolean isBottomListSheetExpanded) {
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/nearby/NearbyMapFragment.java]: Codacy reports that this method is unused, could you double check? "Avoid unused private methods such as 'updateMapCameraAccordingToBottomSheet(boolean)'."

Comment (ashishkumar468) [reply app/src/main/java/fr/free/nrw/commons/nearby/NearbyMapFragment.java]: Yes actually this method got obsolete after fix for #2018 .Although it was a doing a good job adjusting the job based on some calculations for adjusting map zoom level based on bottom sheet's state, but yes can be removed now.

**Comment by neslihanturan** [comment]
- Enormous number of pin problem 
- Autorefresh problem
- Distances
- Forgotten javadocs 

Are fixed, but since I changed a lot this should be tested carefully. Possible test cases:
- Move while map is on
- Go out of nearby borders and see if map is updated nearby pins

**Comment by neslihanturan** [comment]
Ready to be tested @misaochan :)

**Comment by misaochan** [comment]
This works well for me, thanks! :) The only issue now is that I found that rotating the screen makes the map auto-refresh, and then it NEVER loads. Not even when rotating it back to portrait, or after going to Contributions and back to Nearby. 

I will merge this PR first, please fix this in your next PR. Logs below:

```
12-06 03:01:28.282 19338-21801/fr.free.nrw.commons D/NearbyPlaces: 33 results at radius: 2.617924
12-06 03:01:28.300 19338-21801/fr.free.nrw.commons D/NearbyPlaces: https://query.wikidata.org/#SELECT%0A%20%20%20%20%20(SAMPLE(%3Flocation)%20as%20%3Flocation)%0A%20%20%20%20%20%3Fitem%0A%20%20%20%20%20(SAMPLE(COALESCE(%3Fitem_label_preferred_language%2C%20%3Fitem_label_any_language))%20as%20%3Flabel)%0A%20%20%20%20%20(SAMPLE(%3FclassId)%20as%20%3Fclass)%0A%20%20%20%20%20(SAMPLE(COALESCE(%3Fclass_label_preferred_language%2C%20%3Fclass_label_any_language%2C%20%22%3F%22))%20as%20%3Fclass_label)%0A%20%20%20%20%20(SAMPLE(COALESCE(%3Ficon0%2C%20%3Ficon1))%20as%20%3Ficon)%0A%20%20%20%20%20(SAMPLE(COALESCE(%3Femoji0%2C%20%3Femoji1))%20as%20%3Femoji)%0A%20%20%20%20%20%3FwikipediaArticle%0A%20%20%20%20%20%3FcommonsArticle%0A%20%20%20%20%20(SAMPLE(%3FCommons_category)%20as%20%3FCommons_category)%0A%20%20%20WHERE%20%7B%0A%20%20%20%20%20%23%20Around%20given%20location...%0A%20%20%20%20%20SERVICE%20wikibase%3Aaround%20%7B%0A%20%20%20%20%20%20%20%3Fitem%20wdt%3AP625%20%3Flocation.%0A%20%20%20%20%20%20%20bd%3AserviceParam%20wikibase%3Acenter%20%22Point(153.0408%20-27.4531)%22%5E%5Egeo%3AwktLiteral.%0A%20%20%20%20%20%20%20bd%3AserviceParam%20wikibase%3Aradius%20%224.24%22%20.%20%23%20Radius%20in%20kilometers.%0A%20%20%20%20%20%7D%0A%0A%20%20%20%20%20%23%20...%20and%20without%20an%20image.%0A%20%20%20%20%20MINUS%20%7B%3Fitem%20wdt%3AP18%20%5B%5D%7D%0A%0A%20%20%20%20%20%23%20Get%20the%20label%20in%20the%20preferred%20language%20of%20the%20user%2C%20or%20any%20other%20language%20if%20no%20label%20is%20available%20in%20that%20language.%0A%20%20%20%20%20OPTIONAL%20%7B%3Fitem%20rdfs%3Alabel%20%3Fitem_label_preferred_language.%20FILTER%20(lang(%3Fitem_label_preferred_language)%20%3D%20%22en%22)%7D%0A%20%20%20%20%20OPTIONAL%20%7B%3Fitem%20rdfs%3Alabel%20%3Fitem_label_any_language%7D%0A%0A%20%20%20%20%20%23%20Get%20Commons%20category%20(P373)%0A%20%20%20%20%20OPTIONAL%20%7B%20%3Fitem%20wdt%3AP373%20%3FCommons_category.%20%7D%0A%0A%20%20%20%20%20%23%20Get%20the%20class%20label%20in%20the%20preferred%20language%20of%20the%20user%2C%20or%20any%20other%20language%20if%20no%20label%20is%20available%20in%20that%20language.%0A%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%3Fitem%20p%3AP31%2Fps%3AP31%20%3FclassId.%0A%20%20%20%20%20%20%20OPTIONAL%20%7B%3FclassId%20rdfs%3Alabel%20%3Fclass_label_preferred_language.%20FILTER%20(lang(%3Fclass_label_preferred_language)%20%3D%20%22en%22)%7D%0A%20%20%20%20%20%20%20OPTIONAL%20%7B%3FclassId%20rdfs%3Alabel%20%3Fclass_label_any_language%7D%0A%0A%20%20%20%20%20%20%20%23%20Get%20icon%0A%20%20%20%20%20%20%20OPTIONAL%20%7B%20%3FclassId%20wdt%3AP2910%20%3Ficon0.%20%7D%0A%20%20%20%20%20%20%20OPTIONAL%20%7B%20%3FclassId%20wdt%3AP279*%2Fwdt%3AP2910%20%3Ficon1.%20%7D%0A%20%20%20%20%20%20%20%23%20Get%20emoji%0A%20%20%20%20%20%20%20OPTIONAL%20%7B%20%3FclassId%20wdt%3AP487%20%3Femoji0.%20%7D%0A%20%20%20%20%20%20%20OPTIONAL%20%7B%20%3FclassId%20wdt%3AP279*%2Fwdt%3AP487%20%3Femoji1.%20%7D%0A%0A%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%3FwikipediaArticle%20%20%20schema%3Aabout%20%3Fitem%20%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20schema%3AisPartOf%20%3Chttps%3A%2F%2Fen.wikipedia.org%2F%3E%20.%0A%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%3FwikipediaArticle%20%20%20schema%3Aabout%20%3Fitem%20%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20schema%3AisPartOf%20%3Chttps%3A%2F%2Fen.wikipedia.org%2F%3E%20.%0A%20%20%20%20%20%20%20%20%20%20%20SERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22en%22%20%7D%0A%20%20%20%20%20%20%20%20%20%7D%0A%0A%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%3FcommonsArticle%20%20%20schema%3Aabout%20%3Fitem%20%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20schema%3AisPartOf%20%3Chttps%3A%2F%2Fcommons.wikimedia.org%2F%3E%20.%0A%20%20%20%20%20%20%20%20%20%20%20SERVICE%20
    wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22en%22%20%7D%0A%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%7D%0A%20%20%20%7D%0A%20%20%20GROUP%20BY%20%3Fitem%20%3FwikipediaArticle%20%3FcommonsArticle%0A
12-06 03:01:28.307 19338-21801/fr.free.nrw.commons I/System.out: (HTTPLog)-Static: isSBSettingEnabled false
    (HTTPLog)-Static: isSBSettingEnabled false
12-06 03:01:28.423 19338-21804/fr.free.nrw.commons D/NearbyPlaces: 131 results at radius: 4.235801
12-06 03:01:28.423 19338-21804/fr.free.nrw.commons D/NearbyController: Sorting places by distance...
12-06 03:01:28.425 19338-19338/fr.free.nrw.commons D/NearbyFragment: Populating nearby places
12-06 03:01:28.464 19338-19338/fr.free.nrw.commons D/NearbyFragment: Map fragment already exists, just update the map and list
12-06 03:01:28.944 19338-21796/fr.free.nrw.commons D/NearbyPlaces: Reading from query result...
12-06 03:01:29.694 19338-21796/fr.free.nrw.commons D/NearbyPlaces: 800 results at radius: 4.235801
12-06 03:01:29.694 19338-21796/fr.free.nrw.commons D/NearbyController: Sorting places by distance...
12-06 03:01:29.698 19338-19338/fr.free.nrw.commons D/NearbyFragment: Populating nearby places
12-06 03:01:29.799 19338-19338/fr.free.nrw.commons D/NearbyFragment: Map fragment already exists, just update the map and list
12-06 03:01:33.146 19338-21799/fr.free.nrw.commons D/NearbyPlaces: Reading from query result...
12-06 03:01:34.415 19338-21799/fr.free.nrw.commons D/NearbyPlaces: 559 results at radius: 4.235801
12-06 03:01:34.415 19338-21799/fr.free.nrw.commons D/NearbyController: Sorting places by distance...
12-06 03:01:34.424 19338-19338/fr.free.nrw.commons D/NearbyFragment: Populating nearby places
12-06 03:01:34.506 19338-19338/fr.free.nrw.commons D/NearbyFragment: Map fragment already exists, just update the map and list
12-06 03:01:38.163 19338-21801/fr.free.nrw.commons D/NearbyPlaces: Reading from query result...
12-06 03:01:38.549 19338-21801/fr.free.nrw.commons D/NearbyPlaces: 275 results at radius: 4.235801
12-06 03:01:38.549 19338-21801/fr.free.nrw.commons D/NearbyController: Sorting places by distance...
12-06 03:01:38.551 19338-19338/fr.free.nrw.commons D/NearbyFragment: Populating nearby places
12-06 03:01:38.615 19338-19338/fr.free.nrw.commons D/NearbyFragment: Map fragment already exists, just update the map and list
```

**Pull request merged**